packages/effect/src/internal/cause.ts

```diff
@@ -990,10 +993,18 @@ class PrettyError extends globalThis.Error implements Cause.PrettyError {
         : "",
       this.span
     )
+    if (typeof originalError === "object" && originalError !== null) {
+      Object.keys(originalError).forEach((key) => {
+        if (!(key in this)) {
+          // @ts-expect-error
+          this[key] = originalError[key]
+        }
+      })
+    }
   }
 
   get span(): Span | undefined {
```

Comment:
We could just make this a property I guess. Then we don't need `this.#cause`.

Reply:
made the change but apparently there seems to be an issue, I pushed a failing test where spans are not propagated not sure why, I'm going to take a flight will check when I am back if you don't figure it out before